tensorizer/serialization.py

```diff
@@ -616,25 +684,23 @@ def _read_numpytensors(
 
                 # Read our hashes in. We need to read the hashes size, then
                 # read the hash bytes.
-                #
-                # TODO: Actually verify the hashes on request.
                 hashes_begin = shape_end
-                hashes_sz_slice = headers[hashes_begin : hashes_begin + 2]
+                hashes_sz_slice = headers[hashes_begin: hashes_begin + 2]
                 hashes_sz = struct.unpack("<H", hashes_sz_slice)[0]
-                hashes_end = hashes_begin + hashes_sz
-
-                hashes_slice = headers[hashes_begin + 2 : hashes_end]
+                hashes_end = hashes_begin + hashes_sz + 2
+                hashes_slice = headers[hashes_begin + 2: hashes_end]
                 hashes = self._decode_hashes(hashes_slice)
+                self._metadata[name]["hashes"] = hashes
 
                 # Finally, get the tensor data length.
-                data_length = struct.unpack("<q", headers[header_len - 8 :])[0]
+                data_length = struct.unpack("<q", headers[header_len - 8:])[0]
 
                 # Check if the name is in our pre-filtered list of keys
                 # from the class-level filter_func, and then verify
                 # that it passes the method-level filter_func.
                 # Skip it if it fails either check.
                 if name not in self.keys() or (
-                    filter_func is not None and not filter_func(name)
+                        filter_func is not None and not filter_func(name)
                 ):
                     self._file.seek(data_length, io.SEEK_CUR)
                     continue
```

Comment:
Reading hashes needs to be done after the filter check, as `self._metadata[name]` won't exist for pre-filtered entries.

Reply:
This has been addresssed.